lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure.hs

```diff
@@ -245,6 +248,9 @@ getBIP32Paths :: WalletState -> [Read.Address] -> [BIP32Path]
 getBIP32Paths w =
     mapMaybe $ Address.getBIP32Path (addresses w) . Read.toRawAddress
 
+signTxBody :: Write.TxBody -> [BIP32Path] -> WalletState -> Write.Tx
```

Comment:
Or well — at least `Maybe Write.Tx` as `rootSignKey` might be nothing.

Reply:
🤔 For the pure version, I would definitely add explicit failure.

I would also leave out the `[BIP32Path]` argument, because that can be computed from the combination of `TxBody` and `WalletState`. (My feeling is that this argument would be on topic for a function whose abstraction is at the level of `XPrv`, e.g.  `signTxBody :: XPrv → [BIP32Path] → …`, and does not know about the `WalletState`.)

How about this:

```hs
-- | Add signatures for *all* inputs if possible, `Nothing` otherwise.
signTxBody :: Write.TxBody -> WalletState -> Maybe Write.Tx
```

For the version without `Maybe`, I would probably give it a different name, which makes the semantics more clear:

```hs
-- | Add signatures for as many inputs as possible.
cosignTx :: Write.Tx -> WalletState -> Write.Tx
```

We can have either one or both, but for the beginning, I think that `signTxBody` is a good start.